desc/objectives/objective_funs.py

```diff
@@ -742,6 +747,15 @@ def __init__(
         assert normalize_target in {True, False}
         assert (bounds is None) or (isinstance(bounds, tuple) and len(bounds) == 2)
         assert (bounds is None) or (target is None), "Cannot use both bounds and target"
+        assert callable(loss_function), "Loss function must be callable!"
+        assert (
```

Comment:
this might be too restrictive, eg
``len(getfullargspec(jnp.max).args)`` is 6, but all the other args have default values. Besides, the next line will check if its valid when called with a single argument.